--- a/usr/include/c++/v1/wchar.h
+++ b/usr/include/c++/v1/wchar.h
@@ -138,6 +138,7 @@
 #    if _LIBCPP_HAS_WIDE_CHARACTERS
 #      if defined(__cplusplus) && !defined(_LIBCPP_WCHAR_H_HAS_CONST_OVERLOADS) && defined(_LIBCPP_PREFERRED_OVERLOAD)
 extern "C++" {
+#if !defined(__swift__)
 inline _LIBCPP_HIDE_FROM_ABI wchar_t* __libcpp_wcschr(const wchar_t* __s, wchar_t __c) {
   return (wchar_t*)wcschr(__s, __c);
 }
@@ -190,6 +191,7 @@
 inline _LIBCPP_HIDE_FROM_ABI _LIBCPP_PREFERRED_OVERLOAD wchar_t* wmemchr(wchar_t* __s, wchar_t __c, size_t __n) {
   return __libcpp_wmemchr(__s, __c, __n);
 }
+#endif
 }
 #      endif
 
